sympy/vector/coordsysrect.py

```diff
@@ -98,6 +119,27 @@ def __new__(cls, name, location=None, rotation_matrix=None,
             location = Vector.zero
             origin = Point(name + '.origin')
 
+        if transformation is None:
+            transformation = Tuple(rotation_matrix, location)
+
+        if isinstance(transformation, Tuple):
```

Comment:
I thought a bit over the weekend, if we pass a rotation and/or translation, it needs to be stored in `transformation` and passed to the superclass (SymPy's hashing requirements). If we transform the rotation and/or translation into a `Lambda`, the code will slow down a lot, because when the class is reconstructed all orthogonality checks and computations are performed (at least, I think that's the problem).

We can still pass the transformation equations by wrapping them in a `Lambda`.